### PR TITLE
Internal: Nested Accordion animation duration control [ED-11011]

### PR DESCRIPTION
**Pull request by davseve**

### CODE REVIEW

File: modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js [hunk 1]
```diff
@@ -88,7 +86,7 @@ export default class NestedAccordion extends Base {
 
 		animation = accordionItem.animate(
 			{ height: [ startHeight, endHeight ] },
-			{ duration: ANIMATION_DURATION },
+			{ duration: this.animationDuration() },
```
Comment (TzviRabinovitch) [inline modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js:89]: maybe getanimationDuration ?

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 2]
```diff
@@ -314,6 +314,26 @@ protected function register_controls() {
...
+			'n_accordion_animation_duration',
+			[
+				'label' => esc_html__( 'Animation Duration', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 's', 'ms' ],
+				'default' => [
+					'unit' => 'ms',
+					'size' => 400,
+				],
+				'range' => [
```
Comment (TzviRabinovitch) [inline modules/nested-accordion/widgets/nested-accordion.php:327]: do we have a range for seconds?

Comment (TzviRabinovitch) [reply modules/nested-accordion/widgets/nested-accordion.php:327]: 0-3 ?

File: modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js (hunk 1 above)
Comment (davseve) [reply modules/nested-accordion/assets/js/frontend/handlers/nested-accordion.js:89]: Fixed

File: modules/nested-accordion/widgets/nested-accordion.php (hunk 2 above)
Comment (davseve) [reply modules/nested-accordion/widgets/nested-accordion.php:327]: 0-3 is the second's default value.

**Review by TzviRabinovitch — APPROVED** [review]

File: modules/nested-accordion/widgets/nested-accordion.php [hunk 3]
```diff
@@ -314,6 +314,20 @@ protected function register_controls() {
 			]
 		);
 
+		$this->add_control(
+			'n_accordion_animation_duration',
+			[
+				'label' => esc_html__( 'Animation Duration', 'elementor' ),
+				'type' => Controls_Manager::SLIDER,
+				'size_units' => [ 's', 'ms' ],
```
Comment (rodolphebertozzo) [inline modules/nested-accordion/widgets/nested-accordion.php:322]:
```suggestion
				'size_units' => [ 's', 'ms', 'custom' ],
```

**Review by rodolphebertozzo — CHANGES_REQUESTED** [review]

File: modules/nested-accordion/widgets/nested-accordion.php (hunk 3 above)
Comment (davseve) [reply modules/nested-accordion/widgets/nested-accordion.php:322]: @rodolphebertozzo 
Thank's for noticing.
Great suggestion, This was the original product manager's suggestion.
In this case, the toggle animation is controlled by the Javascript animate API.
The animate API receives data in milliseconds.
Just for curiosity, which other time units a user would like to use?
Minutes? Hours?
ודא

Comment (rami-elementor) [reply modules/nested-accordion/widgets/nested-accordion.php:322]: we have only `s` and `ms`, other using are css functions that return calculations of `s` and `ms`.

Comment (davseve) [reply modules/nested-accordion/widgets/nested-accordion.php:322]: @rami-elementor so should we add custom units to this control?

**Pull request merged**